modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java

```diff
@@ -213,6 +214,24 @@ private void addDefaultGlobalElements(Set<String> moduleGlobalElementsNames) {
   }
 
   private void macroExpandGlobalElements(List<ComponentModel> moduleComponentModels, Set<String> moduleGlobalElementsNames) {
+    if (!hasExplicitConfiguration(applicationModel, extensionModel)
+        && extensionModel.getConfigurationModel(MODULE_CONFIG_GLOBAL_ELEMENT_NAME).isPresent()) {
+      ComponentModel configModel = new ComponentModel.Builder()
```

Comment:
Please correct me if my understanding is not correct, but with this check, the implicit configuration would be created even if there are no operations on the application requiring it, right? I was wondering whether it might be possible to refine the check to create the implicit config only when required, when the following conditions are met
- !hasExplicitConfiguration AND
- extensionModel.getConfigurationModel(MODULE_CONFIG_GLOBAL_ELEMENT_NAME).isPresent() 
- there is at least one operation within the app which requires a configuration (by inspecting its ext model?) but no config-ref is defined for it (implicit config)
Not sure if the last check could be added at this point, but I think that I would avoid creating unnecessary implicit configs which might affect existing apps.

Reply:
+1

Reply:
> with this check, the implicit configuration would be created even if there are no operations on the application requiring it, right?

If the extension model has a configuration with name 'config' it means the xml conector has properties. This config is being added by the declarer.

I understand properties are properties of a configuration. In fact, if the operation does not have a 'config-ref' parameter then properties are not accesible.

So, my reasoning is the following:  if the app has a configuration with name 'config' and it does not have an explicit configuration we already know an implicit config is needed.

Maybe we can discus this again on meet.

Reply:
Thanks @Bardammu for the explanation, I see your point. 
My point is whether it is worth to create the implicit config even if there are no operations within the application that require a config (it might be the case that some operations on the extension require a config and some others don't). We can meet to discuss whether it is worth to consider/handle this scenario.